globals/respond_not_implemented.go

```diff
@@ -3,29 +3,46 @@ package globals
 import "github.com/PretendoNetwork/nex-go"
 
 func RespondNotImplemented(packet nex.PacketInterface, protocolID uint8) {
-	client := packet.Sender()
-	request := packet.RMCRequest()
+	switch packet := packet.(type) {
+	case *nex.HPPPacket:
+		client := packet.Sender()
+		request := packet.RMCRequest()
 
-	rmcResponse := nex.NewRMCResponse(protocolID, request.CallID())
-	rmcResponse.SetError(nex.Errors.Core.NotImplemented)
+		rmcResponse := nex.NewRMCResponse(0, request.CallID())
+		rmcResponse.SetError(nex.Errors.Core.NotImplemented)
 
-	rmcResponseBytes := rmcResponse.Bytes()
+		rmcResponseBytes := rmcResponse.Bytes()
 
-	var responsePacket nex.PacketInterface
-	if packet.Version() == 1 {
-		responsePacket, _ = nex.NewPacketV1(client, nil)
-	} else {
-		responsePacket, _ = nex.NewPacketV0(client, nil)
-	}
+		responsePacket, _ := nex.NewHPPPacket(client, nil)
+
+		responsePacket.SetPayload(rmcResponseBytes)
+
+		client.Server().Send(responsePacket)
+	default:
+		client := packet.Sender()
+		request := packet.RMCRequest()
+
+		rmcResponse := nex.NewRMCResponse(protocolID, request.CallID())
+		rmcResponse.SetError(nex.Errors.Core.NotImplemented)
 
-	responsePacket.SetVersion(packet.Version())
-	responsePacket.SetSource(packet.Destination())
-	responsePacket.SetDestination(packet.Source())
-	responsePacket.SetType(nex.DataPacket)
-	responsePacket.SetPayload(rmcResponseBytes)
+		rmcResponseBytes := rmcResponse.Bytes()
 
-	responsePacket.AddFlag(nex.FlagNeedsAck)
-	responsePacket.AddFlag(nex.FlagReliable)
+		var responsePacket nex.PacketInterface
+		if packet.Version() == 1 {
+			responsePacket, _ = nex.NewPacketV1(client, nil)
+		} else {
+			responsePacket, _ = nex.NewPacketV0(client, nil)
+		}
 
-	client.Server().Send(responsePacket)
+		responsePacket.SetVersion(packet.Version())
+		responsePacket.SetSource(packet.Destination())
+		responsePacket.SetDestination(packet.Source())
+		responsePacket.SetType(nex.DataPacket)
+		responsePacket.SetPayload(rmcResponseBytes)
+
+		responsePacket.AddFlag(nex.FlagNeedsAck)
+		responsePacket.AddFlag(nex.FlagReliable)
+
+		client.Server().Send(responsePacket)
+	}
```

Comment:
Minor thing, but a good chunk of these branches is duplicate code. Such as getting the sender, getting the request, setting the error and payload, etc

To minimize duplicate code those pieces should probably happen outside this check, and only handle the real type-specific stuff in each branch